### PR TITLE
Fuzzy matching attribute arguments to ccget

### PR DESCRIPTION
**Pull request by shivupa**
I always forget the attribute names, and I figured this would be helpful. The example below tries to parse `atomcoord` and `mo_coeff` from the log files instead of the correct `atomcoords` and `mocoeffs`. The `homos` is correct and included just to show it works.

A warning is logged that the argument is being fuzzy matched 

```
WARNING:root:Attribute 'atomcoord' not found, but attribute 'atomcoords' is close. Using 'atomcoords' instead.
WARNING:root:Attribute 'mo_coeff' not found, but attribute 'mocoeffs' is close. Using 'mocoeffs' instead.
Attempting to read data/Gaussian/basicGaussian16/dvb_un_sp.log
atomcoords
array([[[ 0.      ,  0.      ,  0.      ],
        [ 0.      ,  0.      ,  1.421154],
        [ 1.203695,  0.      ,  2.140325],
        [ 2.46209 ,  0.      ,  1.477152],
        [ 2.46209 ,  0.      ,  0.055998],
        [-0.952813,  0.      ,  1.966053],
        [ 1.180146,  0.      ,  3.238735],
        [ 3.414904,  0.      , -0.488901],
        [ 3.717531,  0.      ,  2.293665],
        [ 4.977755,  0.      ,  1.825807],
        [ 5.210853,  0.      ,  0.754892],
        [ 3.55288 ,  0.      ,  3.381684],
        [ 5.836811,  0.      ,  2.506791],
        [-1.255441,  0.      , -0.816513],
        [-1.09079 ,  0.      , -1.904532],
        [-2.515665,  0.      , -0.348656],
        [-2.748763,  0.      ,  0.72226 ],
        [-3.374721,  0.      , -1.02964 ],
        [ 1.258395,  0.      , -0.663174],
        [ 1.281944,  0.      , -1.761584]]])
mocoeffs
[array([[ 7.0027e-01,  3.1280e-02,  9.4000e-04, ..., -0.0000e+00, -2.2800e-03,  2.2000e-04],
       [ 7.0029e-01,  3.1080e-02,  7.1000e-04, ...,  0.0000e+00, -2.0300e-03,  1.1000e-04],
       [ 9.9400e-03, -7.6600e-03, -1.1100e-03, ..., -0.0000e+00,  2.6000e-03, -1.6700e-03],
       ...,
       [ 2.0410e-02, -1.4481e-01, -2.9322e-01, ...,  0.0000e+00,  1.4040e-01, -1.3490e-02],
       [ 5.4900e-03, -3.6130e-02, -2.9737e-01, ...,  0.0000e+00,  2.0780e-02,  1.1360e-02],
       [ 4.0360e-02, -3.0538e-01, -5.3642e-01, ...,  0.0000e+00,  6.1608e-01,  2.9064e-01]]),
 array([[ 7.0031e-01,  3.0970e-02,  9.2000e-04, ..., -0.0000e+00, -2.2500e-03,  2.3000e-04],
       [ 7.0033e-01,  3.0790e-02,  7.0000e-04, ...,  0.0000e+00, -2.0100e-03,  1.2000e-04],
       [ 1.0780e-02, -7.6200e-03, -1.1700e-03, ..., -0.0000e+00,  2.6000e-03, -1.7200e-03],
       ...,
       [ 2.1680e-02, -1.5309e-01, -2.9553e-01, ...,  0.0000e+00,  1.4542e-01, -1.1150e-02],
       [ 5.6800e-03, -3.7310e-02, -2.9524e-01, ..., -0.0000e+00,  2.2160e-02,  9.8600e-03],
       [ 4.1730e-02, -3.1344e-01, -5.3676e-01, ...,  0.0000e+00,  6.1443e-01,  2.8805e-01]])]
homos
array([34, 33], dtype=int32)
Attempting to read data/Gaussian/basicGaussian16/dvb_sp.out
atomcoords
array([[[ 0.269445,  1.410118,  0.      ],
        [-1.064785,  0.920662, -0.      ],
        [-1.325389, -0.457058, -0.      ],
        [-0.269445, -1.410118,  0.      ],
        [ 1.064785, -0.920662,  0.      ],
        [-1.90447 ,  1.627522, -0.      ],
        [-2.364759, -0.813041, -0.      ],
        [ 1.90447 , -1.627522,  0.      ],
        [-0.603688, -2.86996 , -0.      ],
        [ 0.269445, -3.89207 , -0.      ],
        [ 1.355175, -3.742314, -0.      ],
        [-1.681859, -3.090152, -0.      ],
        [-0.074247, -4.933029, -0.      ],
        [ 0.603688,  2.86996 , -0.      ],
        [ 1.681859,  3.090152, -0.      ],
        [-0.269445,  3.89207 , -0.      ],
        [-1.355175,  3.742314, -0.      ],
        [ 0.074247,  4.933029, -0.      ],
        [ 1.325389,  0.457058, -0.      ],
        [ 2.364759,  0.813041, -0.      ]]])
mocoeffs
[array([[ 6.9930e-01,  3.1570e-02, -5.0000e-05, ..., -2.7200e-03,  0.0000e+00,  3.3000e-04],
       [ 6.9929e-01,  3.1360e-02,  1.0000e-05, ..., -3.0800e-03,  0.0000e+00,  2.2000e-04],
       [-2.8020e-02,  4.1300e-03,  7.8000e-04, ..., -0.0000e+00,  0.0000e+00, -8.0000e-05],
       ...,
       [ 2.0710e-02, -1.4604e-01, -4.4799e-01, ...,  1.3245e-01,  0.0000e+00, -1.8140e-02],
       [-5.0000e-03,  3.3080e-02,  7.0026e-01, ..., -5.4166e-01,  0.0000e+00, -6.6700e-03],
       [ 4.4390e-02, -3.3032e-01,  1.8649e-01, ..., -2.2716e-01,  0.0000e+00,  2.6965e-01]])]
homos
array([34], dtype=int32)
```

### CODE REVIEW

**Comment by ATenderholt** [comment]
I think it's a good idea. My only concern is adding another dependency that happens to be GPLv2. Only affects `ccget`, but I think that file would have to also be GPLv2 instead of BSD.

**Comment by shivupa** [comment]
Sorry thought I had commented, but I must not have. I realized that `difflib`from the standard library provides all of the necessary functionality with `get_close_matches` so this now works without any additional dependencies!

**Comment by ATenderholt** [comment]
Nice! Does adding a test make sense? There's an existing case where not enough arguments are passed. So I'm guessing it's possible to write a test that'd fail without this patch but passes with it.

**Comment by langner** [comment]
This is interesting. Please make sure the fuzzing only applies to the script, would be confusing to have this at the library level.

File: cclib/scripts/ccget.py [hunk 1]
```diff
@@ -111,6 +112,13 @@ def ccget():
     attrnames = []
     filenames = []
     for arg in arglist:
+        if arg not in ccData._attrlist:
+            fuzzy_attr = difflib.get_close_matches(arg, ccData._attrlist, n=1, cutoff=0.85)
```
Comment (berquist) [inline cclib/scripts/ccget.py:116]: Do you think 0.85 is a good cutoff, or can it be looser?

**Pull request merged**